src/submission/forms.py

```diff
@@ -432,3 +438,25 @@ def __init__(self, *args, **kwargs):
     class Meta:
         model = models.Article
         fields = ('projected_issue',)
+
+
+def utility_clean_orcid(form, field_name='orcid'):
+    """
+    Utility function that cleans an ORCID ID.
+    """
+    orcid = form.cleaned_data[field_name]
+
+    if orcid:
+        orcid_regex = re.compile('([0]{4})-([0-9]{4})-([0-9]{4})-([0-9]{3})([0-9X]{1})')
+        result = orcid_regex.search(orcid)
+
+        if result:
+            orcid = result.group(0)
+        else:
+            form.add_error(
+                field_name,
+                'An ORCID must be in the pattern https://orcid.org/0000-0000-0000-0000 or'
+                ' 0000-0000-0000-0000',
+            )
+
+    return orcid
```

Comment:
can this function take a string instead of a form and a field_name? It'd be easier to test and more handy to use elswhere.
Speaking of which, a unit test would be good too.

The retrieval and error handling could be done in the form's `clean_*` method

Reply:
Sure - will add.